src/v/test_utils/registered_urls.cc

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "registered_urls.h"
+
+#include "vassert.h"
+
+namespace {
+static const auto not_found = http_test_utils::response{
+  .body = "not found", .status = ss::httpd::reply::status_type::not_found};
+
+constexpr std::string_view default_content = "DEFAULT";
+constexpr std::string_view default_response = "canned-response";
+} // namespace
```

Comment:
nit: static is redundant with the anonymous namespace. you could remove static or remove the anonymous namespace (and make the string views static).

Reply:
I had a look at other code and saw a `namespace internal` that I liked so I used it here, although I understand that the compiler won't prevent outside code from just using it and it is more of a convention?

What is preferable of these three:
1. namespace internal/namespace detail
2. anonymouse namespace
3. static

Reply:
there are two cases:

1. code that is "private", but still needs to be in headers. in this case it's common to see `namespace internal` or `namespace details`.
2. code that is scoped to the current file / compilation unit. in this case you can use `static`, or anonymous namespace. Both achieve the same result.

Your code here fits into case 2, so either `static` or anonymous namespace is what you want. By using `namespace internal` symbols could still be available from object files / exported.

Reply:
thanks for the explanation, I used anonymous namespace.